### ISSUE TITLE
Did you find a solution?

### ISSUE THREAD

**Issue by TNAJanssen** [baronerosso/asrock-x570-taichi-5700xt#1]
I have the same issues, but bit different hardware:

3900x
asus x470 prime pro
asus strix rx 5700

Sometimes it boots to a black screen, and sometimes i can login but it freezes after +-15 min.

It worked perfect until 2 days ago :(

**Comment by baronerosso** [comment]
Yes, I find a solution or ... maybe I find it!

Premise, the suspicion was that by itself, MacOS could not always identify the video card; System Report-> Hardware-> PCI was always empty.

The idea was to indicate the device to MacOS, acting in DeviceProperties, and things also change in system report.
![resoconto](https://user-images.githubusercontent.com/3634995/105454951-0b00b900-5c83-11eb-93e5-efb09a5808a8.jpg)

I start using Hackintool -> PCIe, list by Device Name and take a screenshot related to Navi, then click on Export and open it with ProperTree; keep only the entries regarding Navi (4 entries in my case) deleting other voices.
![screen](https://user-images.githubusercontent.com/3634995/105455004-2370d380-5c83-11eb-8291-c5816c7d663e.jpg)

Add device-id (Data) and vendor-id (Data) to each entry with the data detected by Hackintool, copy devices in your config.plist DeviceProperties->Add
![prop](https://user-images.githubusercontent.com/3634995/105455106-4307fc00-5c83-11eb-8eb6-b2a801432926.jpg)

Reboot and … no black screen and faster startup procedure ... for the moment


[config.plist.zip](https://github.com/baronerosso/asrock-x570-taichi-5700xt/files/5853984/config.plist.zip)

**Comment by baronerosso** [comment]
inconclusive solution; it happens less but ... it happens

**Comment by baronerosso** [comment]
Ok, I find a solution for my black screen problem; it’s not an orthodox solution, but … it works, for the moment.

I discovered we can solve the problem restarting WindowServer process, so I create a new user with admin privileges (I called it "reset”, but this is a personal choice) and i build a simple AppleScript to add in login element.

The script is very simple:

do shell script "killall -HUP WindowServer" password "password" with administrator privileges

insert “reset” password and save it as application and then add to user “reset” elements login.

Now, you can use it in case of black screen; you can do a blind login into “reset” account.
Killing WindowsServer force a user logout and you are redirect in login windows.

**Comment by sgufa** [comment]
Hi. Today i've updated to the latest Lilu, WEG and VirtualSMC. Removed the 4 pci references from deviceproperties. 
After that i've rebooted to recovery and done the following from shell:

mount -uw /Volumes/[DISKNAME]
rm -rf /Volumes/[DISKNAME]/Library/Caches/*
rm -rf /Volumes/[DISKNAME]/System/Library/Caches/*
rm -rf /Volumes/[DISKNAME]/System/Library/CacheDelete/*
shutdown -r now

Finally i updated to BigSur 11.2 and for now it's working. I've done several reboots and no black screen.
P.S. I didnt touch PCIe switch setting in bios (lt's in default setting).

**Comment by baronerosso** [comment]
mnmmm, I think it isn't a cache problem.
For what it's worth today I did 15 reboots after modifying PCIe x16 switch and I never had a blackscreen

**Comment by sgufa** [comment]
Did you upgrade any kext?

**Comment by baronerosso** [comment]
I use the last release for all my kext, OC 0.66
These are the ones I currently use:
Kexts:
	AMDRyzenCPUPowerManagement.kext
	AppleALC.kext
	AppleMCEReporterDisabler.kext
	IntelBluetoothFirmware.kext
	Lilu.kext
	SmallTreeIntel82576.kext
	SMCAMDProcessor.kext
	USBMap.kext
	VirtualSMC.kext
	WhateverGreen.kext

**Comment by sgufa** [comment]
The latest Lilu, WEG and VirtualSMC are 3 days old. Try upgrading those files, upgrade big sur to 11.2 and restore the PCIe x16 switch. I can confirm it’s still working and it never worked this way before. Before those operations I got black screen after few reboots. BTW i've just updated my bios to v4.0 and it still works without problems.

**Comment by sgufa** [comment]
do not mind. the latest cold start gave me black screen :(
now I've enabled that PCIe x16 switch setting as in your pic.
it's working for now...

**Comment by sgufa** [comment]
Today black screen on cold start. Restored Pcie x16 switch to Auto and then OK. It's very random, BAH

**Comment by baronerosso** [comment]
yes, PCIe switch does not solve. It could also be the driver of the card not working very well

**Comment by kuaner** [comment]
same issue 
any update?

**Comment by baronerosso** [comment]
> same issue
> any update?

no news at this moment

**Comment by kuaner** [comment]
my solution for now

enable ssh login in settings/share

sudo killall -HUP WindowServer

**Comment by baronerosso** [comment]
[https://forum.amd-osx.com/index.php?threads/blackscreen-handcrafted-solution.1507/#post-7729](https://forum.amd-osx.com/index.php?threads/blackscreen-handcrafted-solution.1507/#post-7729)

in this post I suggest an handcrafted solution


-------
Ok, I find a solution for my black screen problem; it’s not an orthodox solution, but … it works, for the moment.

I discovered we can solve the problem restarting WindowServer process, so I create a new user with admin privileges (I called it "reset”, but this is a personal choice) and i build a simple AppleScript to add in login element.

The script is very simple:

AppleScript:
`do shell script "killall -HUP WindowServer" password "password" with administrator privileges`

insert “reset” password and save it as application and then add to user “reset” elements login.

Now, you can use it in case of black screen; you can do a blind login into “reset” account.
Killing WindowsServer force a user logout and you are redirect in login windows.
---------

**Comment by sgufa** [comment]
Seems that Bigsur 11.3 update solved the issue

**Comment by baronerosso** [comment]
I solved before 11.3 with SSDT-PLUG.aml

**Comment by sgufa** [comment]
can you explain?

> I solved before 11.3 with SSDT-PLUG.aml

**Comment by baronerosso** [comment]
I add SSDT-PLUG.aml in EFI, the one in my posted EFI is modified for AMD

--
il baronerosso via Android

Il sab 1 mag 2021, 15:08 sgufa ***@***.***> ha scritto:

**Comment by sgufa** [comment]
> I add SSDT-PLUG.aml in EFI, the one in my posted EFI is modified for AMD
> […](#)
> -- il baronerosso via Android Il sab 1 mag 2021, 15:08 sgufa ***@***.***> ha scritto:
> can you explain? I solved before 11.3 with SSDT-PLUG.aml 

can you ecplain what that patch does?

**Comment by baronerosso** [comment]
> > I add SSDT-PLUG.aml in EFI, the one in my posted EFI is modified for AMD
> > […](#)
> > -- il baronerosso via Android Il sab 1 mag 2021, 15:08 sgufa _**@**_.***> ha scritto:
> > can you explain? I solved before 11.3 with SSDT-PLUG.aml
> 
> can you ecplain what that patch does?

the .aml affects the motherboard controls, using it will also show "Enable Power Nap" in Energy Saver in System Preferences.

Honestly, I find it in a ryzentosh forum, I realize it has something to do with the behavior of the motherboard and trie it with a pleasant surprise.
How he manages to avoid the blackscreen I don't know;
I know that with other cards they avoid the blackscreen by disabling the serial ports from the bios, but in the Taichi bios ... there is no this option because we do not have serial ports available (even if one appears in Network Preferences)
